src/dma/mod.rs

```diff
@@ -524,63 +585,49 @@ where
     }
 
     #[inline(always)]
-    fn set_common_interrupts(&mut self, interrupts: DmaCommonInterrupts) {
-        unsafe { Self::st() }.cr.modify(|_, w| {
-            w.tcie()
-                .bit(interrupts.transfer_complete)
-                .htie()
-                .bit(interrupts.half_transfer)
-                .teie()
-                .bit(interrupts.transfer_error)
-                .dmeie()
-                .bit(interrupts.direct_mode_error)
-        });
-    }
-
-    #[inline(always)]
-    fn common_interrupts(&self) -> DmaCommonInterrupts {
-        let cr = unsafe { Self::st() }.cr.read();
-        DmaCommonInterrupts {
-            transfer_complete: cr.tcie().bit_is_set(),
-            half_transfer: cr.htie().bit_is_set(),
-            transfer_error: cr.teie().bit_is_set(),
-            direct_mode_error: cr.dmeie().bit_is_set(),
+    fn listen(&mut self, interrupts: DmaCommonInterrupts) {
+        unsafe {
+            Self::st()
+                .cr
+                .modify(|r, w| w.bits(r.bits() | interrupts.bits() as u32));
```

Comment:
because of the existence of "from_bits_retain" value of unused bit is not guaranteed, that mean you need to "delete" those bit before the bitwise "or" operation

Reply:
you want to say that `from_bits_retain` must be `unsafe`?

Reply:
> you need to "delete" those bit before the bitwise "or" operation

tried to fix in last commit